app/src/main/res/layout/fragment_signup.xml

```diff
@@ -86,16 +77,33 @@
             style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox.ExposedDropdownMenu"
             android:layout_width="280dp"
             android:layout_height="wrap_content"
-            android:layout_marginBottom="16dp"
+            android:layout_marginBottom="10dp"
             android:hint="@string/select_role_text">
 
             <AutoCompleteTextView
                 android:id="@+id/role"
                 android:layout_width="match_parent"
-                android:layout_height="match_parent"
+                android:layout_height="50dp"
                 android:inputType="none" />
         </com.google.android.material.textfield.TextInputLayout>
 
+        <TextView
+            android:id="@+id/textView4"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginBottom="10dp"
+            android:text="@string/clientid_text" />
+
+        <EditText
+            android:id="@+id/clientId"
```

Comment:
As I have said previously maybe it would be best to only have a username, email and password field. Up to you, I think if the contact was created previously we don't even need the roles list.